cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -706,6 +706,25 @@ public void updateAppSpec(ApplicationId appId, ApplicationSpecification spec) th
         GSON.toJson(updated));
   }
 
+  /**
+   * Updates the scm metadata of an application.
+   *
+   * @param appId {@link ApplicationId} of the app
+   * @param scmMeta {@link SourceControlMeta} the new scm metadata (git filehash)
+   * @throws IOException when the update operation fails
+   */
+  public void updateAppScmMeta(ApplicationId appId, SourceControlMeta scmMeta) throws IOException {
+    ApplicationMeta existing = getApplication(appId);
+    if (existing == null) {
+      throw new IllegalArgumentException("Application " + appId + " does not exist");
```

Comment:
this shouldn't be IllegalArgumentException, ApplicaitonNotFoundException would be more specific. 

Though I don't think throwing an exception is actually the behavior we want anyway. For example, if somebody deletes an application while the bulk push is happening, we probably just want to skip it instead of failing.